File: backEnd/microservice1/src/main/java/esprit/microservice1/entities/Payment.java
package esprit.microservice1.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Date paymentDate;
    private String paymentMethod;

    @OneToOne
    @JoinColumn(name = "facture_id", referencedColumnName = "id")
    private Facture facture;
}

File: microfront/src/app/khalil/models/payment.ts
import { Facture} from './facture';
import { PAYMETHOD } from './PAYMETHOD';

export interface Payment {
  id?: number;
  paymentDate: Date;
  paymentMethod: PAYMETHOD;
  
  carte: string | null;   
  titulairecarte: string | null;   
  cvv: string | null;      
  jusqua: string | null;   
 

  facture?: Facture;
}
